Sources/MapboxNavigation/NavigationMapView+BuildingHighlighting.swift

```diff
@@ -15,13 +15,122 @@ extension NavigationMapView {
      - returns: Bool indicating if number of buildings found equals number of coordinates supplied.
      */
     @discardableResult public func highlightBuildings(at coordinates: [CLLocationCoordinate2D], in3D extrudesBuildings: Bool = true) -> Bool {
-        return false
+        let _ = coordinates.map({buildingIdentifier(at: $0, in3D: extrudesBuildings) })
+        DispatchQueue.main.asyncAfter(deadline: .now() + handleAsyncTime, execute: {self.addBuildingsLayer(with: self.foundBuildingIds, in3D: extrudesBuildings)})
+        return foundBuildingIds.count == coordinates.count
     }
     
     /**
      Removes the highlight from all buildings highlighted by `highlightBuildings(at:in3D:)`.
      */
     public func unhighlightBuildings() {
-        
+        guard let _ = try? mapView.style.getLayer(with: IdentifierString.buildingExtrusionLayer, type: FillExtrusionLayer.self).get() else { return }
+        foundBuildingIds = Set<Double>()
+        let _ = mapView.style.removeStyleLayer(forLayerId: IdentifierString.buildingExtrusionLayer)
     }
+
+    private func featureIdentifierToDouble(features: [Feature]) -> Double? {
+        if let featureIdentifier = features.first?.identifier {
+            switch featureIdentifier {
+            case .string(let stringId): return Double(stringId)
+            case .number(let numberId):
+                switch numberId {
+                case .double(let doubleId): return Double(doubleId)
+                case .int(let intId): return Double(intId)
+                }
+            }
+        }
+        return nil
+    }
+
+    private func buildingIdentifier(at coordinate: CLLocationCoordinate2D, in3D extrudesBuildings: Bool = true) {
+        let screenCoordinate = mapView.point(for: coordinate, in: self)
+        if let identifiers = try? mapView.__map.getStyleLayers().compactMap({$0.id}).filter({ $0.contains("building") }) {
+            mapView.visibleFeatures(at: screenCoordinate,
+                                    styleLayers: Set(identifiers),
+                                    completion: { [weak self] result in
+                                        guard let validSelf = self else { return }
+                                        if case .success(let features) = result {
+                                            if let identifier = self?.featureIdentifierToDouble(features: features) {
+                                                self?.foundBuildingIds.insert(identifier)
```

Comment:
Instead of adding `foundBuildingIds` as member variable this method should provide callback, which will be called whenever `visibleFeatures()` is done searching for building identifiers. Public method `highlightBuildings()` will have to be updated to call that callback as well, instead of returning boolean value synchronously.

Reply:
Because the `mapView.visibleFeatures(at: styleLayers:,completion:` has an escaping closure. And it only allow the feature querying at one point each time. But  the SDK should allow users to choose multiple waypoints by longPress, and these buildings should be highlighted at the same time after chosen. When dealing with multiple points, we'd better collect all the building IDs from the `mapView.visibleFeatures(at: styleLayers:,completion:` result, and do the extrusion layer filter by these IDs.

Reply:
So you mean that completion handler in `visibleFeatures()` will be called multiple times (depending on number of `identifiers`)? If so we can provide something similar to `DispatchGroup` to wait till all operations complete, and only after that call our completion callback.

Reply:
Yes. The function of `visibleFeatures()` and `func buildingIdentifier(at:)` will be called multiple times, each time it only handles one waypoint. I'll try to use `DispatchGroup ` to handle this. Thanks for the suggestion!